crates/bevy_gltf/src/lib.rs

```diff
@@ -121,18 +125,86 @@ pub struct GltfNode {
     pub extras: Option<GltfExtras>,
 }
 
+impl GltfNode {
+    /// Create a node extracting name and index from gltf def
+    pub fn new(
+        node: &gltf::Node,
+        children: Vec<GltfNode>,
+        mesh: Option<Handle<GltfMesh>>,
+        transform: bevy_transform::prelude::Transform,
+        extras: Option<GltfExtras>,
+    ) -> Self {
+        Self {
+            index: node.index(),
+            name: node.name().map(|s| s.to_string()),
+            children,
+            mesh,
+            transform,
+            extras,
+        }
+    }
+
+    /// Computed name for a node - either a user defined name or a generated name from index
+    pub fn name(&self) -> String {
```

Comment:
I would prefer those to not return owned string, but not sur that's easy

Reply:
I can make name/asset_label be precomputed on construction - but that mean it won't update if the object is mutated. That's probably okay tho.

Reply:
Added a commit with that option, can revert it.